src/main/java/org/broadinstitute/hellbender/utils/downsampling/ReservoirDownsampler.java

```diff
@@ -113,6 +123,18 @@ public boolean hasFinalizedItems() {
         return ! reservoir.isEmpty();
     }
 
+    /**
+     * @param readSeed If supplied, the start position is added to the {@Link Utils#getGatkDefaultRandomSeed} to
+     *                 reset the random seed for the random generator. This is used to ensure that the resivoir
+     *                 downsampler is deterministic in its downsampling.
+     */
+    public List<GATKRead> consumeFinalizedItems(final GATKRead readSeed) {
+        // Use the start position for the read as a seed for the random generator
+        randomGenerator.setSeed(Utils.getGatkDefaultRandomSeed() +
+                (ReadUtils.readHasNoAssignedPosition(readSeed) ? 0 : readSeed.getContig().hashCode() << 32 + readSeed.getStart()));
+        return consumeFinalizedItems();
```

Comment:
Calculating a seed from positional information should probably belong in `PositionalDownsampler`, since `ReservoirDownsampler` is position-agnostic. The method can be passed a seed directly, and its up to the caller to produce it.

Regarding the seed, what's the reason for adding it to the default random seed? Just making it the hash of the position would be sufficient, wouldn't it?

Finally, it might be simpler to separate this logic from `consumeFinalizedItems()`, i.e. have a `setSeed()` method that callers can use as needed.

Reply:
^ I elected to add it to the default random seed as if there were some reason down the line that we would want to configure the default random seed in the gatk that it would still be accounted for here. As for the other two I agree with you.